lib/connection.js

```diff
@@ -905,45 +905,62 @@ function _setClient(conn, client, options, dbName) {
  * Closes the connection
  *
  * @param {Boolean} [force] optional
+ * @param {Boolean} [closeAndClean] optional
  * @param {Function} [callback] optional
  * @return {Promise}
  * @api public
  */
 
-Connection.prototype.close = function(force, callback) {
+Connection.prototype.close = function(force, closeAndClean, callback) {
   if (typeof force === 'function') {
     callback = force;
     force = false;
   }
 
+  if (typeof closeAndClean === 'function') {
+    callback = closeAndClean;
+    closeAndClean = false;
+  }
+
   if (force != null && typeof force === 'object') {
     this.$wasForceClosed = !!force.force;
   } else {
     this.$wasForceClosed = !!force;
   }
 
+  if (closeAndClean !== null) {
```

Comment:
This condition wouldn't be matched if `closeAndClean` is undefined.
```suggestion
  if (closeAndClean != null) {
```